client/signup/steps/domains/index.jsx

```diff
@@ -278,7 +275,7 @@ class DomainsStep extends React.Component {
 				},
 				this.getThemeArgs()
 			),
-			Object.assign( { domainItem }, shouldHideFreePlanItem, useThemeHeadstartItem )
+			Object.assign( { domainItem }, { shouldHideFreePlan }, useThemeHeadstartItem )
```

Comment:
This can be further simplified to
```js
Object.assign( { domainItem, shouldHideFreePlan }, useThemeHeadstartItem )
```
or even
```js
{ domainItem, shouldHideFreePlan, useThemeHeadstartItem }
```

Reply:
I think its best to address in a seperate PR @jsnajdr 🙂 
Will get on this after merging the current PR if that's ok with you.

Reply:
I agree that a separate PR is better :+1: This one is already big enough and as we've seen, a bit fragile and risky.